Dockerfile

```diff
@@ -1,85 +1,23 @@
 FROM debian:buster-slim
 
-# Toolchain dependencies for Unvanquished, Daemon, external_deps, or build-release
-ENV TOOLCHAIN_DEPS=' \
-    autoconf \
-    cmake \
-    curl \
-    g++ \
-    g++-mingw-w64-i686 \
-    g++-mingw-w64-x86-64 \
-    git \
-    libtool \
-    make \
-    p7zip-full \
-    python2 \
-    python3-jinja2 \
-    python3-yaml \
-    rsync \
-'
-
-# Libraries statically linked into the Linux binaries of Daemon
-ENV DAEMON_LINUX_STATIC_DEPS=' \
-    libfreetype6-dev \
-    libgmp-dev \
-    libjpeg-dev \
-    libogg-dev \
-    libopus-dev \
-    libopusfile-dev \
-    libpng-dev \
-    libvorbis-dev \
-    libwebp-dev \
-    nettle-dev \
-    zlib1g-dev \
-'
-
-# These are needed for building Daemon or its dependencies but not statically linked into the binary.
-# The display-related ones are relevant for SDL configuration and are taken from
-# https://wiki.libsdl.org/FAQLinux#How_do_I_get_all_the_dependencies_for_building_SDL_on_Ubuntu.3F
-# The audio ones are relevant for OpenAL configuration.
-ENV DAEMON_LINUX_AV_DEPS=' \
-    libasound2-dev \
-    libaudio-dev \
-    libgl1-mesa-dev \
-    libjack-jackd2-dev \
-    libpulse-dev \
-    libx11-dev \
-    libxcursor-dev \
-    libxext-dev \
-    libxi-dev \
-    libxinerama-dev \
-    libxrandr-dev \
-    libxss-dev \
-    libxxf86vm-dev \
-'
+ARG rev
+RUN test -n "$rev"
 
-RUN apt-get update && apt-get install -y $TOOLCHAIN_DEPS $DAEMON_LINUX_STATIC_DEPS $DAEMON_LINUX_AV_DEPS
+ARG targets='linux-amd64 linux-arm64 linux-armhf windows-i686 windows-amd64 vm'
 
-RUN update-alternatives --set i686-w64-mingw32-gcc /usr/bin/i686-w64-mingw32-gcc-posix && \
-    update-alternatives --set i686-w64-mingw32-g++ /usr/bin/i686-w64-mingw32-g++-posix && \
-    update-alternatives --set x86_64-w64-mingw32-gcc /usr/bin/x86_64-w64-mingw32-gcc-posix && \
-    update-alternatives --set x86_64-w64-mingw32-g++ /usr/bin/x86_64-w64-mingw32-g++-posix
+COPY docker-helper /
+RUN /docker-helper install-system-compiler
```

Comment:
Avoid dividing the package manager fetches into multiple docker steps. All APT commands should be in a single step so that the packages come from a consistent snapshot of the package list.

Reply:
There is now a single `apt-get update` command (which fetches the snapshot  of the package list).

There is now a single `apt-get install` too, which is not required to have a consistent snapshot, but makes things faster.